### PR TITLE
feat: support legacy decorators for typescript

### PR DESCRIPTION
**Pull request by LukeCSamuel-RV**
Addresses #27 

Feedback welcome

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "jiti",
-  "version": "1.6.4",
+  "version": "1.7.0",
```
Comment (pi0) [inline package.json:3]: This number is auto increased by build script 
```suggestion
  "version": "1.6.4",
```

File: src/jiti.ts [hunk 2]
```diff
@@ -80,6 +80,12 @@ export default function createJITI (_filename: string = process.cwd(), opts: JIT
     try { return nativeRequire.resolve(id, options) } catch (e) {}
   }
 
+  let tsConfig = { compilerOptions: {} as any }
+  const tsConfigPath = tryResolve('tsconfig')
```
Comment (pi0) [inline src/jiti.ts:84]: Jiti, by design, acts independently of the environment to have predictive results. We cannot depend on tsconfig.json or babelrc

File: src/babel.ts [hunk 3]
```diff
@@ -13,12 +13,17 @@ export default function transform (opts: TransformOptions): TRANSFORM_RESULT {
       [require('@babel/plugin-transform-modules-commonjs'), { allowTopLevelThis: true }],
       [require('babel-plugin-dynamic-import-node'), { noInterop: true }],
       [require('babel-plugin-transform-import-meta')],
-      [require('@babel/plugin-syntax-class-properties')]
+      [require('@babel/plugin-proposal-class-properties'), { loose: true }]
```
Comment (pi0) [inline src/babel.ts:16]: Is this change necessary for decorators' support? (if yes, please provide an example) Since modern environments already support class properties, we only add syntax support.

Comment (LukeCSamuel-RV) [reply src/babel.ts]: I was following Babel docs which recommended `proposal` over `syntax`, but these may be out of date.  Verified that `syntax` appears to be compatible with the decorators proposal, so I've switched it back.

**Review by pi0 — APPROVED** [review]
Thanks!
